aiidalab_launch/util.py

```diff
@@ -41,13 +42,24 @@ def spinner(
     """Display spinner only after an optional initial delay."""
 
     def spin() -> None:
+
+        # Don't show spinner if verbose output is enabled
+        level = logging.getLogger().getEffectiveLevel()
+        show_spinner = (
+            True if level == logging.NOTSET or level >= logging.ERROR else False
+        )
         if msg:
-            click.echo(f"{msg.rstrip()} ", nl=False, err=True)
-        with click_spinner.spinner():  # type: ignore
+            newline = not show_spinner
+            click.echo(f"{msg.rstrip()} ", nl=newline, err=True)
+
+        if show_spinner:
+            with click_spinner.spinner():  # type: ignore
+                stop.wait()
+            click.echo(
+                (final or "done.") if (completed.is_set() and msg) else " ", err=True
```

Comment:
I believe the line could be slightly simplified, if the default value for `final` was set to `"done."`
```suggestion
                final if (completed.is_set() and msg) else " ", err=True
```

Reply:
Good idea. Done. :-)